Plugin/Magento/Rewards/Controller/Cart/RemoveActionPlugin.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2017-2020 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Plugin\Magento\Rewards\Controller\Cart;
+
+/**
+ * Plugin for {@see \Magento\Reward\Controller\Cart\Remove} used to add AJAX support
+ */
+class RemoveActionPlugin
+{
+    /**
+     * @var \Bolt\Boltpay\Helper\Config Bolt configuration helper instance
+     */
+    private $configHelper;
+
+    /**
+     * @param \Bolt\Boltpay\Helper\Config $configHelper Bolt configuration helper instance
+     */
+    public function __construct(\Bolt\Boltpay\Helper\Config $configHelper)
+    {
+        $this->configHelper = $configHelper;
+    }
+
+    /**
+     * Adds support for AJAX calls by removing redirect from response for AJAX requests
+     *
+     * @see \Magento\Reward\Controller\Cart\Remove::execute
+     *
+     * @param \Magento\Reward\Controller\Cart\Remove        $subject the frontend controller wrapped by this plugin
+     * @param void|\Magento\Framework\App\ResponseInterface $result original result of the wrapped method
+     *
+     * @return void|\Magento\Framework\App\ResponseInterface original result of the wrapped method
+     */
+    public function afterExecute(\Magento\Reward\Controller\Cart\Remove $subject, $result)
```

Comment:
Why we need this?

Reply:
I'll expand my comment in the code.  Which was 

> Adds support for AJAX calls by removing redirect from response for AJAX requests

To expand on this, what we are doing is greatly improving the UX experience.  We are reusing components which are designed for the Shopping Cart page.  The design on that page for reward points removal is to send a location header to force the page to reload.  This is ok from the shopping cart page, but we do not want this from the mini-cart.  We use ajax for a smoother user experience and remove the location header so that the page does not randomly reload the user's page.  This is consistent with other mini-cart edits.

Reply:
Comment updated.

Reply:
Got it. thank you. So looks like we change UX experience for M2 module.
Maybe it makes sense to add this into code comments.